components/script/dom/node.rs

```diff
@@ -2726,18 +2779,109 @@ impl NodeMethods for Node {
     }
 
     // https://dom.spec.whatwg.org/#dom-node-comparedocumentposition
-    fn CompareDocumentPosition(&self, other: &Node) -> u16 {
+    fn CompareDocumentPosition(&self, other: Option<&Node>) -> u16 {
         // step 1.
-        if self == other {
+        if Some(self) == other {
             return 0;
         }
 
-        // FIXME(emilio): This will eventually need to handle attribute nodes.
+        // step 2
+        let node1: Option<&Node> = other;
```

Comment:
The type signatures are unnecessary here

Reply:
I got compiler errors when I took them off the attrs, but they are unnecessary on the nodes and removed now.